src/compute/src/logging/compute.rs

```diff
@@ -245,23 +248,36 @@ pub fn construct<A: Allocate>(
                                     }
                                 }
                             }
-                            ComputeEvent::SourceFrontier(dataflow, source_id, logical) => {
-                                // we should record the source frontier here only if
-                                // there is a corresponding active dataflow. This behavior
-                                // arises because SourceFrontier events are generated by a
-                                // dataflow inspect_container operator, which may outlive
-                                // the corresponding trace or sink recording in the
-                                // current ComputeState until Timely eventually drops it.
-                                let dataflow_key = (dataflow, worker);
-                                if let Some(_) = active_dataflows.get(&dataflow_key) {
-                                    let source_map = storage_sources
-                                        .entry(dataflow_key)
-                                        .or_insert_with(HashMap::new);
-                                    let time_entry = source_map
-                                        .entry(source_id)
-                                        .or_insert((VecDeque::new(), HashMap::new()));
-                                    let time_deque = &mut time_entry.0;
-                                    time_deque.push_back((logical, time.as_nanos()));
+                            ComputeEvent::SourceFrontier(dataflow, source_id, logical, delta) => {
```

Comment:
I'd be curious to know the frequency of these events per source. If it's relatively high, we should scope further optimizing this dataflow, for example by compacting in-place.

Reply:
`SourceFrontier` events are emitted when a `persist_source` frontier advances, which happens when a `ListenEvent::Progress` is generated from `persist`. As far as I could read through the code, the latter events are generated by a storage shard after processing a batch of updates. I am not sure I can get a clear handle on the timing characteristics of this batching process, but my high-level understanding is that we would not want to batch updates too often so as to control the overheads of writing to storage. That is, the frequency here should align with emission of storage control information regarding upper frontiers, not with the entire flow of updates coming through the data plane.